roda-core/roda-core/src/main/java/org/roda/core/index/IndexModelObserver.java

```diff
@@ -988,6 +1002,16 @@ public ReturnWithExceptions<Void, ModelObserver> preservationMetadataCreated(Pre
     if (PreservationMetadataType.EVENT.equals(type)) {
       indexPreservationEvent(pm).addTo(ret);
     } else if (PreservationMetadataType.AGENT.equals(type)) {
+      try {
+        StoragePath binaryPath = ModelUtils.getPreservationMetadataStoragePath(pm);
+        Path path = model.getStorage().getDirectAccess(binaryPath).getPath();
+        BasicFileAttributes attr = Files.readAttributes(path, BasicFileAttributes.class);
+        FileTime fileTime = attr.creationTime();
+        pm.setCreationDate(new Date(fileTime.toMillis()));
```

Comment:
Extract the duplicate code to a private method in this class, which receives the storagePath and returns a Date